src/acme/pd/Path.java

```diff
@@ -45,10 +48,28 @@ public void setBlocksBetweenPickupandDropoff(int blocksBetweenPickupandDropoff)
    
 
     public ArrayList<String> getDeliveryInstructions() {
-        // TODO implement delivery instructions
         ArrayList<String> instructions = new ArrayList<>();
-        Random rand = new Random();
-        IntStream.range(0, 10).forEach(i -> instructions.add("Go " + Direction.values()[rand.nextInt() % 4] + " 1 block"));
+        if (this.path != null && this.path.size() > 0) {
+            for (int i = 0; i < this.path.size() - 1; i++) {
+                instructions.add("Go from " + this.path.get(i).getIntersectionName() + " to "
```

Comment:
DIdn't Harbert say specifically that he wanted E/W/N/S directions in the instructions? Maybe this should say something like "Go (direction) on (roadName) to (IntersectionName)"

Reply:
I guess I can do that